jina/executors/indexers/vector/ngt.py

```diff
@@ -0,0 +1,65 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from typing import Tuple
+
+import numpy as np
+from .numpy import NumpyIndexer
+
+
+class NGTIndexer(NumpyIndexer):
+    """NGT powered vector indexer
+
+    For more information about the NGT supported parameters and installation problems, please consult:
+        - https://github.com/yahoojapan/NGT
+
+    .. note::
+        NGT package dependency is only required at the query time.
+        Quick Install : pip install ngt
+    """
+
+    def __init__(self, num_threads: int = 2, metric: str = 'L2', epsilon: int = 0.1, *args, **kwargs):
+        """
+        Initialize an NGT Indexer
+        :param num_threads: Number of threads to build index
+        :param metric: Should be one of {L1,L2,Hamming,Jaccard,Angle,Normalized Angle,Cosine,Normalized Cosine}
+        :param epsilon: Toggle this variable for speed vs recall tradeoff.
+                        Higher value of epsilon means higher recall
+                        but query time will increase with epsilon
+        """
+
+        super().__init__(*args, **kwargs)
+        self.metric = metric
+        self.index_path = 'index'
+        self.num_threads = num_threads
+        self.epsilon = epsilon
+
+    def get_query_handler(self):
+        """Index all vectors , if already indexed return NGT Index handle """
+
+        import ngtpy
+        vecs = super().get_query_handler()
+        if vecs is not None:
+            ngtpy.create(path=self.index_path, dimension=self.num_dim, distance_type=self.metric)
+            _index = ngtpy.Index(self.index_path)
```

Comment:
Looking at https://github.com/yahoojapan/NGT/blob/master/python/ngt/base.py it seems that ngtpy.create() already returns an index object.

Reply:
@JoanFM Yea we just gotta define the dimensions and path which are mandatory parameters. metrics too is optional but it can be tuned to use needs.

Reply:
@JoanFM ngtpy.create() doesn't return any object. I tested it again , NoneType Object is being created when returning with ngtpy.create()

Reply:
An Index object is only created with ngtpy.index() with the defined index_path.

Reply:
Ok, you are right, sorry for the inconvinience, it seemed to me a strange interface.

Reply:
@JoanFM Man np. It was weird for me too that they wouldn't return the handler for the index object